private/buf/bufformat/formatter.go

```diff
@@ -19,23 +19,33 @@ import (
 	"fmt"
 	"io"
 	"sort"
-	"strconv"
 	"strings"
+	"unicode/utf8"
 
-	"github.com/jhump/protocompile/ast"
+	"github.com/bufbuild/protocompile/ast"
 	"go.uber.org/multierr"
 )
 
 // formatter writes an *ast.FileNode as a .proto file.
 type formatter struct {
-	writer   io.Writer
-	fileNode *ast.FileNode
-	indent   int
+	writer      io.Writer
+	fileNode    *ast.FileNode
+	indent      int
+	lastWritten rune
 
 	// Used to determine how the current node's
 	// leading comments should be written.
 	previousNode ast.Node
 
+	// if true, a space will be written to the output unless the next character
+	// written is a newline (don't wait errant trailing spaces)
+	pendingSpace bool
+	// if true, the formatter is in the middle of printing compact options
+	inCompactOptions bool
+
+	pendingIndent int
+	inline        bool
```

Comment:
It might be nice to add comments to these - there are some inline comments around use of defer that might make sense to move here?

Reply:
done